contracts/GitcoinPassportDecoder.sol

```diff
@@ -0,0 +1,181 @@
+   * @param userAddress User's address
+   * @param schemaUID The UID of the schema used to make the user's attestation
+   */
+  // getPassport(address): Calls GitcoinResolver to get the passport, then uses the provider mapping to decode the bits in the provider bitmap. This function can handle any bitmap version. The version is stored in the attestation, and the contract has all the data for historical bitmap versions. This should also demux the dates and hashes, so that the user gets back a normal looking passport object.
+  function getPassport(address userAddress, bytes32 schemaUID) public view returns (Credential[] memory) {
+    // Set the GitcoinResolver
+    GitcoinResolver resolver = GitcoinResolver(gitcoinResolverAddress);
+
+    // Get the attestation UID from the user's attestations
+    bytes32 attestationUID = resolver.userAttestations(userAddress, schemaUID);
+
+    // Get the attestation from the user's attestation UID
+    Attestation memory attestation = getAttestation(attestationUID);
+
+    // Set up the variables to assign the attestion data output to
+    uint256[] memory providers;
+    bytes32[] memory hashes;
+    uint64[] memory issuanceDates;
+    uint64[] memory expirationDates;
+    uint16 providerMapVersion;
+
+    // Decode the attestion output
+    (providers, hashes, issuanceDates, expirationDates, providerMapVersion) = abi.decode(
+      attestation.data,
+      (uint256[], bytes32[], uint64[], uint64[], uint16)
+    );
+
+    // Set up the variables to record the bit and the index of the credential hash
+    uint256 bit;
+    uint256 hashIndex = 0;
+
+    // Set the list of providers to the provider map version
+    string[] memory mappedProviders = providerVersions[providerMapVersion];
+
+    // Now we iterate over the providers array and check each bit that is set
+    // If a bit is set
+    // we set the hash, issuanceDate, expirationDate, and provider to a Credential struct
+    // then we push that struct to the passport storage array
+
+    // Set the passport array to be returned to equal the length of the passport saved to storage (for the duration of the function call)
+    Credential[] memory passportMemoryArray = new Credential[](hashes.length);
+
+    // Populate the passportMemoryArray
+    for (uint256 i = 0; i < providers.length; ) {
+      bit = 1;
+      uint256 provider = uint256(providers[i]);
+      for (uint256 j = 0; j < 256; ) {
+        // Check that the provider bit is set
+        // The provider bit is set --> set the provider, hash, issuance date, and expiration date to the stuct
+        if (provider & bit > 0) {
+          Credential memory credential;
+          uint256 mappedProvidersIndex = i * 256 + j;
+
+          if (mappedProvidersIndex < mappedProviders.length) {
+            credential.provider = mappedProviders[mappedProvidersIndex];
+          }
+
+          if (hashIndex < hashes.length) {
```

Comment:
No more need for these kind of verifications, if the suggestion above have been implemented